python/hsfs/engine/python.py

```diff
@@ -778,12 +770,10 @@ def _get_app_options(self, user_write_options={}):
             spark_job_configuration=spark_job_configuration,
         )
 
-    def _wait_for_job(self, job, user_write_options=None):
+    def _wait_for_job(self, job, await_termination=True):
         # If the user passed the wait_for_job option consider it,
         # otherwise use the default True
-        while user_write_options is None or user_write_options.get(
-            "wait_for_job", True
-        ):
+        while await_termination:
```

Comment:
Ok, if we have to do it like this, one more thing how to make the code a bit cleaner.
Leave this method as it is and just change it in the `Job.run()` by simply wrapping the parameter in a dict? so we don't have to have the if statement spread through the code?

Reply:
i could do it, but it impacts only 1 place, while run is called in 2 places (where we have to do the check)